mesonbuild/utils/universal.py

```diff
@@ -1413,12 +1414,19 @@ def Popen_safe(args: T.List[str], write: T.Optional[str] = None,
     # If write is not None, set stdin to PIPE so data can be sent.
     if write is not None:
         stdin = subprocess.PIPE
-    if not sys.stdout.encoding or encoding.upper() != 'UTF-8':
-        p, o, e = Popen_safe_legacy(args, write=write, stdin=stdin, stdout=stdout, stderr=stderr, **kwargs)
-    else:
-        p = subprocess.Popen(args, universal_newlines=True, encoding=encoding, close_fds=False,
-                             stdin=stdin, stdout=stdout, stderr=stderr, **kwargs)
-        o, e = p.communicate(write)
+
+    try:
+        if not sys.stdout.encoding or encoding.upper() != 'UTF-8':
+            p, o, e = Popen_safe_legacy(args, write=write, stdin=stdin, stdout=stdout, stderr=stderr, **kwargs)
+        else:
+            p = subprocess.Popen(args, universal_newlines=True, encoding=encoding, close_fds=False,
+                                 stdin=stdin, stdout=stdout, stderr=stderr, **kwargs)
+            o, e = p.communicate(write)
+    except OSError as oserr:
+        if oserr.errno == errno.ENOEXEC:
+            raise MesonException(f'Failed running {args[0]!r}, binary or interpreter not executable.\n'
+                                 'Possibly wrong archiecture or the executable bit is not set.')
```

Comment:
```suggestion
                                 'Possibly wrong architecture or the executable bit is not set.')
```

Reply:
I only change that typo and now flake is unhappy…

```
mesonbuild/cmake/interpreter.py:55:5: F401 '.common.CMakeConfiguration' imported but unused
```

😕

Reply:
It's not unhappy! You rebased against git master so it runs fine...

Reply:
Huh ok, no idea whats going on, Github sent me an email with the failure report…